benchmark/http2/inheritance.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const common = require('../common.js');
+const PORT = common.PORT;
+
+const ORIGINAL = 'original';
+const SET_PROTOTYPE_OF = 'setPrototypeOf';
+const PARAMS = 'params';
+
+var bench = common.createBenchmark(main, {
+  streams: [100, 500, 1000, 2500, 5000, 10000],
+  version: [ORIGINAL, SET_PROTOTYPE_OF, PARAMS],
+  requests: [5000, 10000, 20000, 50000]
+}, { flags: ['--expose-http2', '--no-warnings'] });
+
+function main(conf) {
+  const http2 = require('http2');
+  const maxConcurrentStreams = Number(conf.streams);
+  const requests = Number(conf.requests);
+
+  class MyServerRequest extends http2.Http2ServerRequest {
+    getUserAgent() {
+      return this.headers['user-agent'] || 'unknown';
+    }
+  }
+
+  class MyServerResponse extends http2.Http2ServerResponse {
+    status(code, contentType) {
+      this.writeHead(code, { 'Content-Type': contentType });
+    }
+  }
+
+  let server;
+
+  if (conf.version === ORIGINAL) {
+    // Original prototype
+    server = http2.createServer((req, res) => {
+      MyServerResponse.prototype.status.call(res, 200, 'text/plain');
+      res.end(`User Agent: ${MyServerRequest.prototype.getUserAgent.call(req)}`
```

Comment:
The 'original' is slower because you're having to access a prototype of an object from outside of it's scope and then you're using `call` on it. This should just be:

```js
this.writeHead(200, { 'Content-Type': contentType });
res.end(`User Agent: ${this.headers['user-agent'] || 'unknown'}`);
```

Reply:
To expand on this, I think you would actually not even want to run the original test & the prototype test against your modified version of the code. I think those should be tested against master while the `PARAMS` test case should run against your build.